### PR TITLE
Referral History Update

### PR DESCRIPTION
**Pull request by alexnatalia**
# Task

[PW-404]: Title.

## Changes

1. Updated Referral history.
2. Increased Wallet version.

[PW-404]: https://soramitsu.atlassian.net/browse/PW-404

### CODE REVIEW

File: src/services/types.ts [hunk 1]
```diff
@@ -130,3 +136,11 @@ export type ReferrerReward = {
   timestamp: number;
   amount: string;
 };
+
+export type ReferrerSetReferrer = {
```
Comment (Nikita-Polyakov) [inline src/services/types.ts:140]: Shouldn't it be named `ReferrerSetReferral`? I guess Referrer couldn't set another Referrer

File: src/services/types.ts [hunk 2]
```diff
@@ -130,3 +136,11 @@ export type ReferrerReward = {
...
+  referrer: string;
```
Comment (Nikita-Polyakov) [inline src/services/types.ts:141]: I guess you should define in this type this structure of `historyElement` `data`
https://github.com/sora-xor/sora-subquery/blob/develop/src/mappings/setReferralHandler.ts#L16

```
from: string; // referral
to: string; // referrer
```

File: src/services/types.ts [hunk 3]
```diff
@@ -130,3 +136,11 @@ export type ReferrerReward = {
...
+};
+
+export type ReferrerReserve = {
+  amount: string;
```
Comment (Nikita-Polyakov) [inline src/services/types.ts:145]: Could you define all attributes from `historyElement ``data `in this type please? 
https://github.com/sora-xor/sora-subquery/blob/develop/src/mappings/referralReserveHandler.ts#L16

```
from: string;
to: string;
amount: string;
```

File: src/services/subquery/parser.ts [hunk 4]
```diff
@@ -239,6 +248,21 @@ export default class SubqueryDataParser implements ExplorerDataParser {
       // TODO: wait for Subquery support:
       // Operation.Rewards
       // utility.batch
+      case Operation.ReferralSetInvitedUser: {
+        const data = transaction.data as ReferrerSetReferrer;
+        payload.to = data.referrer;
```
Comment (Nikita-Polyakov) [inline src/services/subquery/parser.ts:253]: `payload.to = data.to` ?

**Review by Nikita-Polyakov — APPROVED** [review]

**Review by RustemYuzlibaev — APPROVED** [review]

**Pull request merged**